isy-security/src/main/java/de/bund/bva/isyfact/security/oauth2/client/Authentifizierungsmanager.java

```diff
@@ -2,58 +2,120 @@
+ * for example in a scheduled/background thread and/or in the service-tier.
+ * <p>
+ * The preferred way is to use {@link #authentifiziere(String)} which will automatically
  */
 public interface Authentifizierungsmanager {
 
     /**
-     * Attempts to authorize the client for the given {@code registrationId} via its configured OAuth 2.0 Flow.
-     * After successful authentication the authenticated principal in the {@link org.springframework.security.core.context.SecurityContext}
-     * will be updated.
+     * Attempts to authorize the client for the given {@code oauth2ClientRegistrationId} via its configured OAuth 2.0 Flow.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
+     * <p>
      * The chosen OAuth 2.0 Flow will depend on the authorization grant type configured in the application properties.
      * The currently supported flows are:
      * <ul>
-     *     <li>Client Credentials Flow (grant type: client_credentials)</li>
-     *     <li>Resource Owner Password Credentials Flow (grant type: password)</li>
+     *     <li>Client Credentials (grant type: client_credentials)</li>
+     *     <li>Resource Owner Password Credentials (grant type: password)</li>
      * </ul>
+     * Additional properties for the Resource Owner Password Credentials Flow will be resolved via the matching
+     * registration in {@link IsyOAuth2ClientConfigurationProperties#registration}.
      *
-     * @param registrationId
-     *         ID of the OAuth 2.0 Client to authorize
+     * @param oauth2ClientRegistrationId
+     *         registration ID of the OAuth 2.0 Client to authorize
      * @throws AuthenticationException
      *         if authentication fails
      */
-    void authentifiziere(String registrationId) throws AuthenticationException;
+    void authentifiziere(String oauth2ClientRegistrationId) throws AuthenticationException;
 
     /**
      * Attempts to create and authorize a client with the given credentials via the OAuth 2.0 Client Credentials Flow.
-     * After successful authentication the authenticated principal in the {@link org.springframework.security.core.context.SecurityContext}
-     * will be updated.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
      * <p>
-     * This method is only intended to ease migration to the Client Credentials Flow.
-     * If possible it is preferred to use {@link #authentifiziere(String)} instead.
+     * This method is only intended to allow OAuth 2.0 Client Authentication in cases where the credentials
+     * are obtained externally and cannot be configured in the application properties.
+     * If possible it is strongly preferred to use {@link #authentifiziere(String) authentication with a client registration ID} instead.
```

Comment:
That's not correct. Please take a look in the different authentication constellation, which have to be supported by isy-security:
- **internal authentications:** for example batch, task uses the method `authentifiziere(oAuth2ClientRegistrationId)` and the manually not automatically configured `clientregistration.grant-type` defines, which credential flow will be used, but it has to match with the registered _authentification data_ in IAM. It will not work to use a _client-credential_-flow for authentication, but in IAM this user is still registered per _username/password_.
- **external authentications:** this will always be a valid constellation by receiving a request from an external application - see _isyfact reference architecture_ -, which will processed at first by a _service-gateway_ or in the future by an _authentication-gateway_. The gateway decides depending on the given external request parameters - _username/password_ or _clientId/clientSecret_-, which authentication method it will use in _isy-security_, the `authentifiziereSystem` or `authentifiziereClient`. If the external application will request with the parameters _username/password_ the gateway has to authenticate itself as the _intermediary_ too and uses its local manually configured _clientRegistration authenticationdata_. if the external application will request with the parameters _clientId/clientSecret_ the gateway only has to forward these _authenticationdata_ to the method `authentifiziereClient`, because these credentials will never extists in a local configured _clientRegistration_. Within the _client-credential_-flow there is no need for authentication of the _intermediary_.

In consequence there is **no priority or preference** about the usage of one of the `AuthentifizierungsManager` methods **nor the intention** that one of the methods should _never_ be used.
The only preference is to migrate a registered user in IAM from a _natural user_ to a _technical client_ as soon as possible. But for a _mass user data migration_ in IAM there are other additional prerequisits, which has to be fulfilled first.

Reply:
THe sentence will be changed to:
` If a internal clientRegistration configuration exists, it is strongly preferred to use {@link #authentifiziere(String)...`